rosidl_runtime_c/include/rosidl_runtime_c/u16string_functions.h

```diff
@@ -31,53 +31,156 @@ extern "C"
+ *   or the memory allocation failed
  */
 ROSIDL_GENERATOR_C_PUBLIC
 bool
 rosidl_runtime_c__U16String__init(rosidl_runtime_c__U16String * str);
 
+/// Deallocate the memory of the rosidl_runtime_c__U16String structure.
+/*
+* Calling the function with an already deallocated sequence is a no-op.
+*
+* \param[inout] str a pointer to a U16 string structure to be finalized
+*/
 ROSIDL_GENERATOR_C_PUBLIC
 void
 rosidl_runtime_c__U16String__fini(rosidl_runtime_c__U16String * str);
 
+/// Assign the uint16_t value of n characters to the rosidl_runtime_c__U16String structure.
+/*
+ *
+ * This function returns `false` if memory cannot be allocated,
+ * if the input uint16_t pointer is null or if the size is higher than SIZE_MAX.
+ * In both cases no error message is set.
+ *
+ * \param[inout] str a pointer to a U16 string structure
+ * \param[in] value uint16_t pointer to be assigned
+ * \param[in] n size of the value string
+ * \return true if successful, false if the passed string pointer is null
+ *   or if the passed value pointer is null or if the size is higher than SIZE_MAX
+ *   or if the memory reallocation failed.
+ */
 ROSIDL_GENERATOR_C_PUBLIC
 bool
 rosidl_runtime_c__U16String__assignn(
   rosidl_runtime_c__U16String * str, const uint16_t * value, size_t n);
 
+/// Assign the c string pointer value of n characters to the rosidl_runtime_c__U16String structure.
+/*
+ *
+ * This function is identical to rosidl_runtime_c__U16String__assignn() except the type of the
+ * value is c string pointer
+ *
+ * \see rosidl_runtime_c__U16String__assignn()
+ *
+ * \param[inout] str a pointer to a U16 string structure
+ * \param[in] value c string pointer to be assigned
+ * \param[in] n size of the value string
+ * \return true if successful, false if the passed string pointer is null
+ *   or if the passed value pointer is null or if the size is higher than SIZE_MAX
+ *   or if the memory reallocation failed.
+ */
 ROSIDL_GENERATOR_C_PUBLIC
 bool
 rosidl_runtime_c__U16String__assignn_from_char(
   rosidl_runtime_c__U16String * str, const char * value, size_t n);
 
+/// Assign the uint16_t pointer to the rosidl_runtime_c__U16String structure.
+/*
+ *
+ * This function is identical to rosidl_runtime_c__U16String__assignn() except the length of the
+ * uint16_t does not have to be given.
+ *
+ * \see rosidl_runtime_c__U16String__assignn()
+ *
+ * \param[inout] str a pointer to a rosidl_runtime_c__U16String structure
+ * \param[in] value to be assigned
+ * \return true if successful, false if the passed string pointer is null
+ *   or if the passed value pointer is null or if the size is higher than SIZE_MAX
+ *   or if the memory reallocation failed.
+ */
 ROSIDL_GENERATOR_C_PUBLIC
 bool
 rosidl_runtime_c__U16String__assign(
   rosidl_runtime_c__U16String * str, const uint16_t * value);
 
+/// Get the length of the uint16_t pointer.
+/*
+ * This function returns the length of the input value pointer.
+ *
+ * \param[in] value to calculate the length
```

Comment:
"to calculate the length" doesn't really describe the value. Something like value points to a sequence of 16 bit chars for which the first null char is determined.